### PR TITLE
Support of "transformations" parameter in Delete Resources

### PR DESCRIPTION
**Pull request by RTLcoil**

### CODE REVIEW

File: test/api_spec.coffee [hunk 1]
```diff
@@ -711,7 +711,7 @@ describe "api", ->
   # Replace `it` with  `it.skip` below if you want to disable it.
   it "should list folders in cloudinary", (done)->
     @timeout helper.TIMEOUT_LONG
-    Q.all([
+    return Q.all([
```
Comment (tocker) [inline test/api_spec.coffee]: Why return?

File: test/api_spec.coffee [hunk 2]
```diff
@@ -732,8 +732,8 @@ describe "api", ->
       expect(sub_1.folders[1].path).to.eql('test_folder1/test_subfolder2')
       cloudinary.v2.api.sub_folders('test_folder_not_exists')
     ).then((result)->
-      console.log('error test_folder_not_exists should not pass to "then" handler but "catch"')
```
Comment (tocker) [inline test/api_spec.coffee]: Why remove the message?

File: src/api.coffee [hunk 3]
```diff
@@ -153,24 +153,24 @@ exports.delete_resources = (public_ids, callback, options = {}) ->
   resource_type = options["resource_type"] ? "image"
   type = options["type"] ? "upload"
   uri = ["resources", resource_type, type]
-  call_api("delete", uri, _.extend({"public_ids[]": public_ids}, api.only(options, "keep_original", "invalidate")), callback, options)
+  call_api("delete", uri, _.extend({"public_ids[]": public_ids}, api.only(options, "keep_original", "invalidate", "transformations")), callback, options)
```
Comment (tocker) [inline src/api.coffee]: `transformations` should be an array serialised as eager transformation. I don't see it here.

Also, Please add tests.

File: package.json [hunk 4]
```diff
@@ -27,6 +27,7 @@
     "pretest": "npm run compile",
     "prepublish": "npm run compile",
     "test": "node_modules/.bin/mocha -R spec --recursive test/",
+    "test:watch": "node_modules/.bin/mocha -R spec --watch test/",
```
Comment (tocker) [inline package.json]: Unrelated to the task. Please put in a separate commit.

File: test/api_spec.coffee (hunk 2 above)
Comment (RTLcoil) [reply test/api_spec.coffee]: Seems like a line was removed during the merge but we don't know why (the developer doesn't remember...)

File: test/api_spec.coffee (hunk 1 above)
Comment (RTLcoil) [reply test/api_spec.coffee]: To get the test working.

File: src/api.coffee (hunk 3 above)
Comment (RTLcoil) [reply src/api.coffee]: Transformations are formatted on caller side and just retransferred to Clodinary there, so no data transformations required. We'll check how UT could be added for this case.

File: package.json (hunk 4 above)
Comment (RTLcoil) [reply package.json]: Sure, we'll remove it.

**Comment by RTLcoil** [comment]
Please review the following PR with your comments implemented: https://github.com/cloudinary/cloudinary_npm/pull/178
